### DISCUSSION TITLE
Upgrade issue from 1.28.1 to 2.23.0, following the upgrade process to the letter

### DISCUSSION THREAD

**Discussion by nbaud** [kimai/kimai#5126] [On-Premise: Installation & Updates]
Reposting here as the other issue was somehow blocked/closed for an unknown reason, but my issue is very real. Note that I made a mistake in the version number I reported, though. as it's 2.23.0 and not 2.3. Sorry about that. Maybe you will find that it makes more sense.

I followed the steps in the documentation for upgrading, but stuck on this. I tried some manual tricks but to no avail. I am upgrading from 1.28.1 to 2.23.0 (I know it's a big step)

The main issue is this

[notice] Migrating up to DoctrineMigrations\Version20240926111739
[error] Migration DoctrineMigrations\Version20230327143628 failed during Execution. Error: "The table with name "kimai2.kimai2_working_times" already exists."
                                                                                                                        
 [ERROR] The table with name "kimai2.kimai2_working_times" already exists. 
from

Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [OK]
Executing script assets:install [OK]

nico@gigai1:/var/www/kimai2$ bin/console kimai:update

Kimai updates running ...
=========================

[notice] Migrating up to DoctrineMigrations\Version20240926111739
[error] Migration DoctrineMigrations\Version20230327143628 failed during Execution. Error: "The table with name "kimai2.kimai2_working_times" already exists."
                                                                                                                        
 [ERROR] The table with name "kimai2.kimai2_working_times" already exists.                                              
                                                                                                                        

nico@gigai1:/var/www/kimai2$ bin/console doctrine:migrations:status
+----------------------+----------------------+------------------------------------------------------------------------+
| Configuration                                                                                                        |
+----------------------+----------------------+------------------------------------------------------------------------+
| Storage              | Type                 | Doctrine\Migrations\Metadata\Storage\TableMetadataStorageConfiguration |
|                      | Table Name           | migration_versions                                                     |
|                      | Column Name          | version                                                                |
|----------------------------------------------------------------------------------------------------------------------|
| Database             | Driver               | Doctrine\DBAL\Driver\PDO\MySQL\Driver                                  |
|                      | Name                 | kimai2                                                                 |
|----------------------------------------------------------------------------------------------------------------------|
| Versions             | Previous             | DoctrineMigrations\Version20230126002049                               |
|                      | Current              | DoctrineMigrations\Version20230126002050                               |
|                      | Next                 | DoctrineMigrations\Version20230327143628                               |
|                      | Latest               | DoctrineMigrations\Version20240926111739                               |
|----------------------------------------------------------------------------------------------------------------------|
| Migrations           | Executed             | 62                                                                     |
|                      | Executed Unavailable | 0                                                                      |
|                      | Available            | 70                                                                     |
|                      | New                  | 8                                                                      |
|----------------------------------------------------------------------------------------------------------------------|
| Migration Namespaces | DoctrineMigrations   | /var/www/kimai2/migrations                                             |
+----------------------+----------------------+------------------------------------------------------------------------+
nico@gigai1:/var/www/kimai2$ 
3 replies
@nbaud
nbaud
5 hours ago
Also

nico@gigai1:/var/www/kimai2$ bin/console doctrine:migrations:status
+----------------------+----------------------+------------------------------------------------------------------------+
| Configuration                                                                                                        |
+----------------------+----------------------+------------------------------------------------------------------------+
| Storage              | Type                 | Doctrine\Migrations\Metadata\Storage\TableMetadataStorageConfiguration |
|                      | Table Name           | migration_versions                                                     |
|                      | Column Name          | version                                                                |
|----------------------------------------------------------------------------------------------------------------------|
| Database             | Driver               | Doctrine\DBAL\Driver\PDO\MySQL\Driver                                  |
|                      | Name                 | kimai2                                                                 |
|----------------------------------------------------------------------------------------------------------------------|
| Versions             | Previous             | DoctrineMigrations\Version20230126002049                               |
|                      | Current              | DoctrineMigrations\Version20230126002050                               |
|                      | Next                 | DoctrineMigrations\Version20230327143628                               |
|                      | Latest               | DoctrineMigrations\Version20240926111739                               |
|----------------------------------------------------------------------------------------------------------------------|
| Migrations           | Executed             | 62                                                                     |
|                      | Executed Unavailable | 0                                                                      |
|                      | Available            | 70                                                                     |
|                      | New                  | 8                                                                      |
|----------------------------------------------------------------------------------------------------------------------|
| Migration Namespaces | DoctrineMigrations   | /var/www/kimai2/migrations                                             |
+----------------------+----------------------+------------------------------------------------------------------------+

@nbaud
nbaud
5 hours ago
that part

composer install --no-dev --optimize-autoloader
Went all fine

**Comment by nbaud** [comment]
I also want to add that I managed to go through the upgrade process by adding the doctrines manually where the table already existed, resulting in an error.



```
nico@gigai1:/var/www/kimai2$ bin/console kimai:update

Kimai updates running ...
=========================

[notice] Migrating up to DoctrineMigrations\Version20240926111739
[error] Migration DoctrineMigrations\Version20240214061246 failed during Execution. Error: "The table with name "kimai2.kimai2_access_token" already exists."
                                                                                                                        
 [ERROR] The table with name "kimai2.kimai2_access_token" already exists.                                               
                                                                                                                        

nico@gigai1:/var/www/kimai2$ bin/console doctrine:migrations:version DoctrineMigrations\\Version20240214061246 --add

 WARNING! You are about to add, delete or synchronize migration versions from the version table that could result in data lost. Are you sure you wish to continue? (yes/no) [yes]:
 > yes

 DoctrineMigrations\Version20240214061246 added to the version table.

nico@gigai1:/var/www/kimai2$ bin/console kimai:update

Kimai updates running ...
=========================

[notice] Migrating up to DoctrineMigrations\Version20240926111739
[notice] finished in 81.2ms, used 20M memory, 3 migrations executed, 7 sql queries

                                                                                                                        
 [OK] Successfully migrated to version: DoctrineMigrations\Version20240926111739                                        
                                                                                                                        



 Rebuilding your cache, please be patient ...

 // Clearing the cache for the prod environment with debug false                                                        

                                                                                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          
                                                                                                                        


 // Warming up the cache for the prod environment with debug false                                                      

                                                                                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully warmed.                                           
                                                                                                                        


                                                                                                                        
 [OK] Congratulations! Successfully updated Kimai to version 2.23.0           
 ```

**Comment by nbaud** [comment]
Now, I am just wondering if I didn't do anything bad to the database by adding manually these (two) doctrine migrations. What do you think?

**Reply by kevinpapst** [reply]
Whether or not you will run into issues can only be seen by comparing the expected and the current database structure.
Check the problematic migration files and make sure everything that is in there is in your database.

**Comment by kevinpapst** [comment]
Somehow your migrations (and the database structure) is messed up. And I am certain that you did not share all the executed steps. 
These errors can only show up if you execute the migration multiple times, which I cannot find in your description.

Just rollback your backup and start from scratch.
